Multiple Documentation Enhancements

ashcrow opened the pull request:
- basic ``commissaire-containermgr-service`` docs
- Removed out of date etcd logging docs
- Pointed components to services for service docs
- Added configuration example to ``commissaire-storage-service``

ashcrow committed:
doc: Added basic commissaire-containermgr-service information.

ashcrow committed:
doc: componenets now references services for docs.

ashcrow committed:
doc: Added example configuration for storage service.

ashcrow committed:
doc: Removed etcd logging docs.

ashcrow commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(5) [multi-host-mgr-2] Create ContainerManager Service](https://trello.com/c/hBL4DKWc/553-5-multi-host-mgr-2-create-containermanager-service)

mbarnes commented on doc/services.rst:
```diff
@@ -8,6 +8,19 @@ Commissaire's message bus and provide/consume services.
+Example Configuration File
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+By default ``commissaire-clustermgr-service`` looks at ``/etc/commissaire/containermgr.conf``
+for it's configuration. This can be overriden with the ``-c/--config`` flag.
```
As an aside, `read_config_file()` in `commissaire` still defaults to `/etc/commissaire/commissaire.conf`, which applies to all services if `-c/--config` is not given.  Is that still okay, or do want to bake these service-specific defaults into the services?  Then we could remove the `-c` from the `.service` files.

The documentation as written here isn't quite true.

ashcrow replied:
Ah true. I'll just update to say it defaults to ``/etc/commissaire/commissaire.conf`` and this is the relevant section for this service.

mbarnes commented on doc/services.rst:
```diff
@@ -35,23 +35,22 @@ Commissaire Investigator
+Example Configuration File
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+By default ``commissaire-storage-service`` looks at ``/etc/commissaire/storage.conf``
+for it's configuration. This can be overriden with the ``-c/--config`` flag.
```
Similar concern here.

mbarnes reviewed (approved):
Approving the docs, but suggesting a code change.

ashcrow commented:
:arrow_up:

mbarnes commented:
Hmm, that wasn't quite what I meant.  I should have been clearer.

`commissaire-storage.service` still has `-c /etc/commissaire/storage.conf`.  I'm suggesting using that as the default path in `storage/__init__.py` and remove it from the `.service` file, so that `commissaire-storage-service` actually *does* look at `/etc/commissaire/storage.conf` by default as the docs say.

ashcrow commented:
@mbarnes I follow. OK. I'll update.

ashcrow committed:
config: Added default to read_config_file.

The new default parameter allows for a service to define it's own
fallback location for a configuration file.

mbarnes reviewed (approved):
Yep, this looks good.  I almost forgot that falling back to the default path entails special behavior, so we do need that 2nd argument.

ashcrow commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(2) [multi-host-mgr-2] Allow services to define their own default config file paths](https://trello.com/c/uz8haeZe/604-2-multi-host-mgr-2-allow-services-to-define-their-own-default-config-file-paths)
